etl/models/types/predicate.py

```diff
@@ -0,0 +1,9 @@
+from enum import Enum
+
+
+class Predicate(str, Enum):
+    """An enum of predicates for RDF Triples."""
```

Comment:
Don't need this enum, use pyoxigraph NamedNode with singletons attached to a class.

```
class ArkgSchema:
   TITLE = NamedNode("http://imlapps.github.io/anti-recommender/schema#title")
```

Instances:
```
class ArkgInstances:
   @staticmethod
    def anti_recommendation_iri(record_key: str) -> NamedNode:
        return NamedNode(f"http://imlapps.github.io/anti-recommender/anti-recommendation/{record_key}")
```

Reply:
Instead of minting a predicate suggest using schema.org's CreativeWork predicates e.g., `schema:name` and `schema:url`.

Reply:
https://schema.org/Recommendation

```
<wikipedia page URL> a schema:WebPage .
<wikipedia page URL> schema:title "Whatever ";
<wikipedia page URL> schema:url <http://wikipedia.org...>

<anti-recommendation IRI> a schema:Recommendation .
<anti-recommendation IRI> schema:itemReviewed <wikipedia page URL> .
```

Reply:
Yes, I am using schema.org's predicates now.